File: frontend/src/types/repository.ts
export interface RepositoryMetadata {
  name: string;
  owner: string;
  description: string | null;
  stars: number;
  forks: number;
  language: string | null;
  default_branch: string;
}

export interface RepositoryNode {
  name: string;
  path: string;
  type: "file" | "directory";
  children: RepositoryNode[];
}

export interface RepositorySummaryResponse {
  summary: string;
  metadata: RepositoryMetadata;
  tree: RepositoryNode;
}

File: backend/app/schemas/repository.py
from pydantic import BaseModel

from app.schemas.parser import RepositoryNode


class RepositoryImportRequest(BaseModel):
    url: str


class RepositoryImportResponse(BaseModel):
    status: str
    repository: str


class RepositoryMetadata(BaseModel):
    name: str
    owner: str
    description: str | None
    stars: int
    forks: int
    language: str | None
    default_branch: str


class RepositoryAnalysis(BaseModel):
    metadata: RepositoryMetadata
    local_path: str
    tree: RepositoryNode